src/fidesops/ops/service/connectors/saas/connector_registry_service.py

```diff
@@ -133,3 +138,83 @@ def load_registry(config_file: str) -> ConnectorRegistry:
+    to be run on server bootstrap.
+    """
+    for connector_type in registry.connector_types():
+        logger.debug(
+            "Determining if any updates are needed for connectors of type %s based on templates...",
+            connector_type,
+        )
+        template: ConnectorTemplate = registry.get_connector_template(  # type: ignore
+            connector_type
+        )
+        saas_config_template = SaaSConfig.parse_obj(load_config(template.config))
+        template_version: Union[LegacyVersion, Version] = parse_version(
+            saas_config_template.version
+        )
+
+        connection_configs: Iterable[ConnectionConfig] = ConnectionConfig.filter(
+            db=db,
+            conditions=(ConnectionConfig.saas_config["type"].astext == connector_type),
+        ).all()
+        for connection_config in connection_configs:
+            saas_config_instance = SaaSConfig.parse_obj(connection_config.saas_config)
+            if parse_version(saas_config_instance.version) < template_version:
+                logger.info(
+                    "Updating SaaS config instance '%s' of type '%s' as its version, %s, was found to be lower than the template version %s",
+                    saas_config_instance.fides_key,
+                    connector_type,
+                    saas_config_instance.version,
+                    template_version,
+                )
+                try:
+                    update_saas_instance(
+                        db,
+                        connection_config,
+                        template,
+                        saas_config_instance,
+                    )
+                except Exception:
+                    logger.error(
+                        "Encountered error attempting to update SaaS config instance %s",
+                        saas_config_instance.fides_key,
+                        exc_info=True,
+                    )
+
+
+def update_saas_instance(
+    db: Session,
+    connection_config: ConnectionConfig,
+    template: ConnectorTemplate,
+    saas_config_instance: SaaSConfig,
+) -> None:
+    """
+    Replace in the DB the existing SaaS instance configuration data
+    (SaaSConfig, DatasetConfig) associated with the given ConnectionConfig
+    with new instance configuration data based on the given ConnectorTemplate
+    """
+    template_vals = SaasConnectionTemplateValues(
+        name=connection_config.name,
+        key=connection_config.key,
+        description=connection_config.description,
+        secrets=connection_config.secrets,
+        instance_key=saas_config_instance.fides_key,
+    )
+
+    config_from_template: Dict = load_config_with_replacement(
+        template.config, "<instance_fides_key>", template_vals.instance_key
+    )
+    connection_config = ConnectionConfig.get_connection_config_or_error(
+        db, template_vals.key  # type: ignore
+    )
+    connection_config.update_saas_config(db, SaaSConfig(**config_from_template))
```

Comment:
As you mentioned, if our secrets are missing or invalid now we need to make it very clear somewhere that they need to update this.

Reply:
yup - but in discussions with @galvana, we'd agreed to tackle this in a separate issue/PR (per https://github.com/ethyca/fidesops/issues/1098#issuecomment-1239763318). it does make me a bit uneasy to include this as is, but in the interest of iterating, are you OK with splitting that piece out?

Reply:
makes sense to me!